mcs/class/corlib/System/AppDomainSetup.cs

```diff
@@ -139,14 +140,19 @@ static string GetAppBase (string appBase)
 				appBase = appBase.Substring (7);
 				if (Path.DirectorySeparatorChar != '/')
 					appBase = appBase.Replace ('/', Path.DirectorySeparatorChar);
-				if (Environment.IsRunningOnWindows) {
-					// Under Windows prepend "//" to indicate it's a local file
-					appBase = "//" + appBase;
+			}
+			appBase = Path.GetFullPath (appBase);
+
+			if (Path.DirectorySeparatorChar != '/') {
+				bool isExtendedPath = appBase.StartsWith (@"\\?\", StringComparison.Ordinal);
+				if (appBase.IndexOf (':', isExtendedPath ? 6 : 2) != -1) {
+					throw new NotSupportedException ("The given path's format is not supported.");
 				}
-			} else {
-				appBase = Path.GetFullPath (appBase);
 			}
 
+			// Trigger a validation of the path
+			new FileIOPermission (FileIOPermissionAccess.PathDiscovery, appBase);
```

Comment:
On Windows FEATURE_MONO_CAS doesn't seem to be defined. If I add 
```
#if FEATURE_MONO_CAS
   ...
#endif
```

around the `FileIOPermission()` call it doesn't seem to be included and the test fails. How do I make sure it's enabled on Windows?